client/lxd_instances.go

```diff
@@ -1238,6 +1239,41 @@ func (r *ProtocolLXD) ExecInstance(instanceName string, exec api.InstanceExecPos
 				}
 			}()
 		}
+
+		if exec.RecordOutput {
+			if args.Stdout != nil || args.Stderr != nil {
+				stdoutResultFile, ok := opAPI.Metadata["output"].(map[string]any)["1"].(string)
+				if !ok {
+					return nil, fmt.Errorf("")
+				}
+
+				reader, err := r.GetInstanceExecOutputLogFile(instanceName, filepath.Base(stdoutResultFile))
+				if err != nil {
+					return nil, err
+				}
+
+				if args.Stdout != nil {
+					_, err = io.Copy(args.Stdout, reader)
+					if err != nil {
+						return nil, err
+					}
+				}
+
+				if args.Stderr != nil {
+					_, err = io.Copy(args.Stderr, reader)
+					if err != nil {
+						return nil, err
+					}
+				}
+
+				// After the content has been streamed to one or both of the available handle,
+				// delete the log file.
+				err = r.DeleteInstanceExecOutputLogFile(instanceName, filepath.Base(stdoutResultFile))
+				if err != nil {
+					return nil, err
+				}
+			}
+		}
```

Comment:
I think there are a few issues here. 
* We need to wait for the operation to complete, then refresh `opAPI`. Otherwise `opAPI.Metadata` may not contain the `output` key.
* It looks like the `stdout` file is being copied to both `args.Stderr` and `args.Stdout`.
* Only one of the output files is being deleted.

We should have:
```golang
if exec.RecordOutput && (args.Stdout != nil || args.Stderr != nil) {
    err = op.Wait()
    if err != nil {
        return nil, err
    }

    opAPI = op.Get()
    if args.Stdout != nil {
        // Get file, write to args.Stdout, delete file.
    }

    if args.Stderr != nil {
        // Get file, write to args.Stderr, delete file
    }
}

```

Reply:
yeah I realized that. It should be fixed now